### PR TITLE
Removed references to Docker in Kubernetes API #31393

### LINKED ISSUE #31393: Remove references to Docker in Kubernetes API
H/T @shannonxtreme! 

Ref: https://github.com/kubernetes/website/issues/30939

----

My findings in the one-page autogenerated API reference (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/)

There might be some (or all) that don't apply to the dockershim removal.

Container
File: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2294

Actions: Remove "docker" from the phrase "docker image". Maybe replace with "container image" where it makes sense, or just "image".

ContainerStatus
File: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2582

Actions: Make the description of containerID not docker-specific.

PodStatus
File: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L3835

Actions: Remove sentence about docker inspect from containerStatuses

ContainerStateTerminated
File: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2561

Actions: Make the description of containerId not docker specific.

EphemeralContainer
File: https://github.com/kubernetes/kubernetes/blob/4c9659b1a6c526612534f75140b1e6f345d1efdb/staging/src/k8s.io/api/core/v1/types.go#L3597

Actions: Remove "docker" from the phrase "docker image". Maybe replace with "container image" where it makes sense, or just "image".

/assign @sftim

### PR DESCRIPTION
**Pull request by whtssub**
#### What type of PR is this?
/kind cleanup
/kind deprecation
/kind feature

#### Which issue(s) this PR fixes:

Fixes   <a href="https://github.com/kubernetes/website/issues/31393">#31393</a> 

#### Special notes for your reviewer:
Additional changes can be implemented in docker-specific secrets mentions and further cited examples. 
#### Does this PR introduce a user-facing change?
NONE

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
@SubhasmitaSw looks like you have 2 commits that negate each other. can you please check?

**Comment by SubhasmitaSw** [comment]
@dims actually I accidentally pushed the changes to the master branch and tried to revert back and push again in a new branch, looks it didn't work. Any nudges to resolve this?

**Comment by dims** [comment]
@SubhasmitaSw ,

For this specific problem, please do this.

Step 1: ensure you are still in `master` branch by running `git branch` (there will be `*` in front of the current branch
Step 2: if you don't see `master` as the current branch then run `git checkout master` to switch to master
Step 3: run `git log` to see the commits, you will see the 2 commits `removed references to Docker in Kubernetes API` right at the top
Step 4: now you know that you are in the correct branch and you will be dealing with the commits you want to fix. now run `git reset --hard HEAD~1` to drop just the first commit
Step 5: run `git log` to check if that commit is gone.
Step 6: run `git push -f origin master` to update the PR

Note that these steps are just to recover from your problem with opening a PR with master. Our regular workflow is described here ( https://www.kubernetes.dev/docs/guide/github-workflow/ ) you can hit one of us up on slack if you have a problem in the future.

**Comment by SubhasmitaSw** [comment]
Thanks again @dims, looks like this time it's resolved.

**Comment by dims** [comment]
> Thanks again @dims, looks like this time it's resolved.

yep! thanks!

**Comment by liggitt** [comment]
Changes look good. You'll need to run `make update` to update generated files and docs and check those in as well

**Comment by SubhasmitaSw** [comment]
Hello @liggitt I was wondering if these instructions are for me, then I might need a little context to it. 

> Changes look good. You'll need to run `make update` to update generated files and docs and check those in as well

**Comment by dims** [comment]
@SubhasmitaSw yes. it was for you. if you click on details in the`pull-kubernetes-verify` job you will see the list of things that failed https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/107691/pull-kubernetes-verify/1484879565900746752 

`make update` runs a bunch of scripts in the hack/ directory named `update-` which will fix the problems found by the 3 verify scripts that failed (see the details in url above). one another choice is to run just the 3 scripts since you know which ones failed from that CI job output. in this case you need to run the following 
- hack/update-generated-protobuf.sh
- hack/update-generated-swagger-docs.sh
- hack/update-openapi-spec.sh

So either the `make update` or the 3 scripts will fix up the failures. you can add the files that got updated into another git commit and update this PR.

**Comment by dims** [comment]
@SubhasmitaSw see https://github.com/kubernetes/kubernetes/pull/107706/ for the "lost" command and a commit with the regenerated files as well. feel free to pull those into this PR and then i can close that out.

**Comment by dims** [comment]
@SubhasmitaSw please review https://github.com/kubernetes/kubernetes/pull/107691/commits (you should not have the merge commit and you have duplicate of the commit from me and also remove the etcd file as well)

**Comment by dims** [comment]
@SubhasmitaSw now you are missing the first commit - https://github.com/kubernetes/kubernetes/pull/107706/commits/829698fbe7314e893f37a6cce7cb5ea6ffb00dd7

**Comment by dims** [comment]
@liggitt this is ready as well

/approve
/lgtm

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -2555,7 +2555,7 @@ type ContainerStateTerminated struct {
 	// Time at which the container last terminated
 	// +optional
 	FinishedAt metav1.Time `json:"finishedAt,omitempty" protobuf:"bytes,6,opt,name=finishedAt"`
-	// Container's ID in the format 'docker://<container_id>'
+	// Container's ID in the format '<container_id>'
```
Comment (liggitt) [inline staging/src/k8s.io/api/core/v1/types.go:2558]: based on https://github.com/kubernetes/kubernetes/blob/d4b49be81a28a2de1c63bbd84ceb1bf3660fa983/pkg/kubelet/container/runtime.go#L214-L223, it looks like this is supposed to have the structure `<type>://<container_id>`

```suggestion
	// Container's ID in the format '<type>://<container_id>'
```

File: staging/src/k8s.io/api/core/v1/types.go [hunk 2]
```diff
@@ -2595,7 +2595,7 @@ type ContainerStatus struct {
 	Image string `json:"image" protobuf:"bytes,6,opt,name=image"`
 	// ImageID of the container's image.
 	ImageID string `json:"imageID" protobuf:"bytes,7,opt,name=imageID"`
-	// Container's ID in the format 'docker://<container_id>'.
+	// Container's ID in the format '<container_id>'.
```
Comment (liggitt) [inline staging/src/k8s.io/api/core/v1/types.go:2598]:
```suggestion
	// Container's ID in the format '<type>://<container_id>'.
```

**Review by liggitt — COMMENTED** [review]
sorry, just noticed the `<type>://<container_id>` bit

**Comment by SubhasmitaSw** [comment]
> sorry, just noticed the `<type>://<container_id>` bit

Suggestions updated.

**Comment by dims** [comment]
@SubhasmitaSw looks like you need a bit more files to be fixed before the verify CI job goes green:
https://github.com/kubernetes/kubernetes/commit/91ed93100f3b4db30cf659f8e47272a2715980bc

**Comment by dims** [comment]
@SubhasmitaSw please rebase against master and then squash the commits. That should be enough to get this wrapped up.

**Pull request merged**